### PR TITLE
OD-11 [Feature] Integrates new DS Provider

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-11 https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-11
## Description
Integration DS Provider into Interactive map component
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cY6roI7sof
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/interface/add-markers.js [hunk 1]
```diff
@@ -86,40 +87,55 @@ Fliplet.InteractiveMap.component('add-markers', {
...
       }
+    },
+    initDataSourceProvider(currentDataSourceId) {
+      const $vm = this
+      let dataSourceData = {
+        dataSourceTitle: 'Markers data source',
+        dataSourceId: currentDataSourceId,
+        appId: Fliplet.Env.get('appId'),
+        default: {
+          name: 'Profile data for ' + Fliplet.Env.get('appName'),
```
Comment (tonytlwu) [inline js/interface/add-markers.js:122]:
```suggestion
          name: 'Marker data for ' + Fliplet.Env.get('appName'),
```

File: js/interface/add-markers.js [hunk 2]
```diff
@@ -86,40 +87,55 @@ Fliplet.InteractiveMap.component('add-markers', {
...
+      }
 
-      this.dataSourceId = ds.id
-    }
-  },
-  methods: {
+      this.dataSourceProvider = Fliplet.Widget.open('com.fliplet.data-source-provider', {
+        selector: '#dataSourceProvider',
+        data: dataSourceData,
+        onEvent: function(event, dataSource) {
```
Comment (tonytlwu) [inline js/interface/add-markers.js:132]: Use ES6 arrow functions so the context doesn't change inside the function.

```suggestion
        onEvent: (event, dataSource) => {
```

File: js/interface/add-markers.js [hunk 3]
```diff
@@ -86,40 +87,55 @@ Fliplet.InteractiveMap.component('add-markers', {
...
+          if (event === 'dataSourceSelect') {
+            $vm.markersDataSourceChange(dataSource)
```
Comment (tonytlwu) [inline js/interface/add-markers.js:134]:
```suggestion
            this.markersDataSourceChange(dataSource)
```

File: js/interface/add-markers.js [hunk 4]
```diff
@@ -86,40 +87,55 @@ Fliplet.InteractiveMap.component('add-markers', {
...
+
+          this.dataSourceId = ds.id
+          this.markersDataSource.columns = ds.columns
+
+          this.dataWasChanged = true
         })
       }
+    },
+    initDataSourceProvider(currentDataSourceId) {
+      const $vm = this
```
Comment (tonytlwu) [inline js/interface/add-markers.js:116]:
```suggestion
```

File: widget.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "Interactive graphics",
-  "package": "com.fliplet.interactive-map",
+  "name": "Interactive graphics-andrey",
+  "package": "com.fliplet.interactive-map-andrey",
```
Comment (tonytlwu) [inline widget.json:3]:
```suggestion
  "package": "com.fliplet.interactive-map",
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
@upplabs-alex-levchenko Can you confirm you were involved in the review of this PR and explain why these were missed? I don't think we're introducing any new standards here.

File: widget.json (hunk 5 above)
Comment (AndrRyaz) [reply widget.json:3]: @tonytlwu Sorry this is my bad with this comment, lost this thing

**Comment by AndrRyaz** [comment]
@tonytlwu 
All the comments above have been resolved, sorry for my inattention, in the future I will be more attentive

**Comment by upplabs-alex-levchenko** [comment]
@tonytlwu Yo, why I'm tagged here?

**Comment by tonytlwu** [comment]
@upplabs-alex-levchenko You can see the question I left above https://github.com/Fliplet/fliplet-widget-interactive-map/pull/88#pullrequestreview-513826163 if you scroll up. We're notifying you that there are issues with code standard such as https://github.com/Fliplet/fliplet-widget-interactive-map/pull/88#discussion_r509395470 

We will start rejecting PRs citing non-compliance to Fliplet's coding standards, and will not be reviewing the code logic unless it meets our standard as it is taking up a lot of Fliplet's time to be reviewing code formatting and linting standards when Upplabs has been hired to meet these standards.

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: All interactive graphic instances
Verify: Feature works correctly with the data sources

Squash and merge: Integrates new data source provider

**Pull request merged**